src/modules/Home/components/MainSlider/MainSlider.tsx

```diff
@@ -0,0 +1,168 @@
+import cn from 'classnames';
+import React, {PureComponent} from 'react';
+
+import Arrow from 'assets/images/ImageArrowDown';
+import Flex from 'common/components/Flex/Flex';
+import MainSlide from '../../containers/MainSlideContainer';
+import ProgressCircle from './components/ProgressCircle/ProgressCircle';
+
+import styles from './mainSlider.scss';
+
+interface ISlide {
+	order: number;
+	content: any;
+	name: 'string';
+	title: 'string';
+	smallText: any;
+	className: string;
+}
+
+interface IProps {
+	currentSlideIndex: number;
+	onSlide: any;
+	slideTiming: number;
+	mainSlides: ISlide[];
+	resetMainSlider: any;
+}
+
+interface IState {
+	prevSlideTimer: number;
+	nextSlideTimer: number;
+	slidesTimer: any;
+	isAnimated: boolean;
+}
+
+class MainSlider extends PureComponent<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+		this.state = {
+			isAnimated: false,
+			prevSlideTimer: 0,
+			nextSlideTimer: 0,
+			slidesTimer: undefined,
```

Comment:
what is this? remove

Reply:
you don't need to use constructor() when you want define a state. just write `state= {}`  in body of your class